services/app-api/src/resolvers/questionAnswers/createQuestion.ts

```diff
@@ -0,0 +1,70 @@
+import { MutationResolvers } from '../../gen/gqlServer'
+import { isCMSUser, packageStatus } from '../../domain-models'
+import { logError, logSuccess } from '../../logger'
+import {
+    setErrorAttributesOnActiveSpan,
+    setSuccessAttributesOnActiveSpan,
+} from '../attributeHelper'
+import { ForbiddenError, UserInputError } from 'apollo-server-lambda'
+import { isStoreError, Store } from '../../postgres'
+import { GraphQLError } from 'graphql'
+
+export function createQuestionResolver(
+    store: Store
+): MutationResolvers['createQuestion'] {
+    return async (_parent, { input }, context) => {
+        const { user, span } = context
+
+        if (!isCMSUser(user)) {
+            const msg = 'user not authorized to create a question'
+            logError('createQuestion', msg)
+            setErrorAttributesOnActiveSpan(msg, span)
+            throw new ForbiddenError(msg)
+        }
+
+        // Return error if package is not found or errors
+        const result = await store.findHealthPlanPackage(input.pkgID)
+
+        if (isStoreError(result)) {
+            const errMessage = `Issue finding a package of type ${result.code}. Message: ${result.message}`
+            logError('createQuestion', errMessage)
+            setErrorAttributesOnActiveSpan(errMessage, span)
+            throw new Error(errMessage)
+        }
+
+        if (result === undefined) {
+            const errMessage = `Issue finding a package with id ${input.pkgID}. Message: Result was undefined`
```

Comment:
Quibble: I think this message (and the same one for indexQ) could be more specific to what happened, which is that they tried to ask a question on a package that does not exist. Saying “Result was undefined” is an implementation detail to how we defined the postgres function and not really helpful to someone making the query, I think.